geekodoc/rng/2_5.2/geekodoc-v2.rnc

```diff
@@ -1442,17 +1442,24 @@ include "db52itsxi.rnc"
   }
 
   # personname
+  # Issue #83: Allow givenname, lineage, and honorific
   div {
     # Removed text
     db.personname =
     ## The personal name of an individual
     element personname {
       db.personname.attlist,
-      (db.honorific
-      | db.firstname
-      | db.surname
-      | db.lineage
-      | db.othername)+
+      ( (db.honorific
+          | db.firstname
+          | db.surname
+          | db.lineage
+          | db.othername)+
+       | (db.honorific
+          | db.givenname
+          | db.surname
+          | db.lineage
+          | db.othername)+
+        )
```

Comment:
Is it helpful to have `+` at the end? I guess the question is whether e.g. if your last name is something Spanish, you'd write `<surname>Garcia</surname><surname>Lopez</surname>` or `<surname>Garcia Lopez</surname>`. And I guess it would be the latter, because that's not just less markup but also preempts unwanted name reordering.

Do you think this might work? (It's a bit of an issue that this codifies an order, but I don't know if that can be avoided.)

```suggestion
      ( db.honorific?,
        db.givenname?,
        db.othername?,
        db.surname,
        db.lineage?)
        )
```

Reply:
> Is it helpful to have + at the end?

It's not from me, it's DocBook's content model for `personname` (minus the `db._text` pattern):

<details>
  <summary>Original content model for personname</summary>

```
db.personname =
    ## The personal name of an individual
    element personname {
      db.personname.attlist,
     (db._text
       | (db.honorific
          | db.firstname
          | db.surname
          | db.lineage
          | db.othername)+
       | (db.honorific
          | db.givenname
          | db.surname
          | db.lineage
          | db.othername)+)
    }
```
</details>

The reason for that was to make it to adapt DocBook's content model.


> I guess the question is whether e.g. if your last name is something Spanish, would you write `<surname>Garcia</surname><surname>Lopez</surname>` or `<surname>Garcia Lopez</surname>`?

Names are complicated. DocBook's markup is useful for maybe ~80-90% of all names. Not all names can be marked up appropriate. For example, some Gaelic names are even more difficult for DocBook.

Back to your question: I'm not familiar with Spanish names. It probably depends a bit on how the last name (or surname) is perceived. Are both parts independent from each other? Or are they an integral part and can't be separated? I don't know. :man_shrugging: 


> Do you think this might work? (It's a bit of an issue that this codifies an order, but I don't know if that can be avoided.)

Technically, it does work. If you want to avoid an order, use the interleave pattern (`&`). The stylesheets picks the respective elements, see `common/common.xsl`, named template `person.name`. 

If you add a `@role` attribute to `personname`, it can even pick and arrange the elements to format the name (valid values are `family-given`, `last-first`, and `first-last`). So the order of these elements are quite irrelevant.